src/app.js

```diff
@@ -302,32 +302,23 @@ class Messenger extends EventEmitter {
       attachments
     } = message;
 
-    if (this.options.emitGreetings && this.greetings.test(text)) {
-      const firstName = session.profile && session.profile.first_name.trim() || '';
-      const surName = session.profile && session.profile.last_name.trim() || '';
-      const fullName = `${firstName} ${surName}`;
-
-      this.emit('text.greeting', new Response(this, {event, senderId, session, firstName, surName, fullName}));
-      return;
-    }
-
-    if (this.help.test(text)) {
-      this.emit('text.help', new Response(this, {event, senderId, session}));
+    if (this.emitOptionalEvents(event, senderId, session, text)) {
       return;
     }
 
     if (quickReply) {
-      debug('message.quickReply payload: "%s"', quickReply.payload);
+      const payload = quickReply.payload;
+      debug('message.quickReply payload: "%s"', payload);
 
-      this.emit('text', new Response(this, {event, senderId, session, source: 'quickReply', text: quickReply.payload}));
-      this.emit('message.quickReply', new Response(this, {event, senderId, session, payload: quickReply.payload}));
+      this.emit('text', new Response(this, {event, senderId, session, source: 'quickReply', text: payload, normalizedText: payload.toLowerCase().trim()}));
```

Comment:
The way text is normalized is repeated every time it's used. I'd like to see it become a method like `this.normalizeString(payload)`. This way you could also write tests that followed the README:

> #### Normalized message content
> 
> To help keep application code simple, the SDK makes these guarantees about _normalized text_:
> * it will be lowercase
> * it will be stripped of leading and trailing whitespace